### PR TITLE
Ygo 74

### PR DESCRIPTION
**Pull request by Erakedon**

### CODE REVIEW

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+//import DeckResult from './DeckResult/DeckResult';
+import Axios from 'axios';
+import LoadingIcon from './../../Shared/LoadingIcon/LoadingIcon';
+import SelectFromListField from '../../Shared/SelectFromListField/SelectFromListField';
+import ArchetypeResult from './ArchetypeResult/ArchetypeResult';
+
+class ArchetypeList extends Component {
+    state = { 
+        archetypesList: [],
+
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js]: Remove empty space.

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    componentWillUnmount() {
+        clearTimeout(this.getArchetypeListTimeoutReference);
+     }
+
+    getArchetypeListTimeoutReference = {};
+
+    getData() {
+        this.getArchetypeListTimeoutReference = setTimeout(() => {            
+            const locParam = this.getLoactionParameters();
+
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js]: Remove empty space (at least between variables declarations).

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+    getPageNumber() {
+        if(!this.props.location.search) return 1;
+
+        const locationParameters = this.getLoactionParameters();
+        if(locationParameters.pageNumber)
+        {
+            return Number(locationParameters.pageNumber);
+        }
+        return 1
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js:96]: Add empty space before return.

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+    render() { 
+        let pagesList = [];
+        const currentPage = this.getPageNumber();
+        const pagesOnSidesOfResult = 3;
+
+        for (let i = 1; i <= this.state.numberOfPages; i++) {
+            let classList = i === currentPage ? "pageNumber currentPage" : "pageNumber"
+
+            if(i+1 > currentPage + pagesOnSidesOfResult) {
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js]: Change ```i+1``` to ```i + 1```

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                onClick={() => {
+                    this.searchParametersFromLocation();
+                    this.changeSearchParameter("pageNumber",currentI+"");
+                    this.newSearchPage();
+                    }}>
+                    {i}
+                </div>
+            );
+            
+            if(i-1 < currentPage - pagesOnSidesOfResult - 1) {
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js]: Change ```i-1``` to ```i - 1```

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+    getLoactionParameters() {
+        const locationParameters = this.props.location.search.split('?');
+        let locationConcreteValues = {};
+        locationParameters.forEach(param => {
+            let nameAndValue = param.split("=");
+            if(nameAndValue[0])
+                locationConcreteValues[nameAndValue[0]] = nameAndValue[1];
+        });
+        return locationConcreteValues;
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js:108]: Add empty space before return.

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+            
+            if(i-1 < currentPage - pagesOnSidesOfResult - 1) {
+                pagesList.push(<i className="fas fa-ellipsis-h" key="dots1" />);
+                i = currentPage - pagesOnSidesOfResult;
+            }                
+        }
+
+        let resultList = [];
+        this.state.archetypesResultList.forEach((el) => {
+
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js]: Remove empty space.

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+             />);       
+        });
+
+        const resultsOnPage = this.getLoactionParameters().numberOfResults;
+        if(this.state.errorLoading) resultList = (
+            <div key="err" className="centredFlexContainer">
+                <h1>Error on loading list of archetypes :(</h1>
+            </div>
+        );
+        // else if (!this.state.loadingData && this.state.archetypesResultList.length === 0) resultList = (
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js]: Remove commented code.

File: src/Pages/ArchetypeList/ArchetypeList.js [hunk 9]
```diff
@@ -0,0 +1,241 @@
+import React, { Component } from 'react';
+import SearchPanel from '../../Shared/SearchPanel/SearchPanel';
+//import DeckResult from './DeckResult/DeckResult';
```
Comment (NightAngell) [inline src/Pages/ArchetypeList/ArchetypeList.js]: Remove commented import.

File: src/Pages/ArchetypePage/ArchetypePage.js [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+      let gamesTotal = 0;
+      let gamesWon = 0;  
+
+      let newDeckState = {
+        archetypeId: res.data.archetypeId,
+        name: res.data.name.split("_")[0],
+        archetype: res.data.archetype,
+        cardsList: res.data.cardsInArchetype,
+
+
```
Comment (NightAngell) [inline src/Pages/ArchetypePage/ArchetypePage.js]: Double empty space is forbidden. Remove at least one line of empty space

File: src/Pages/ArchetypePage/ArchetypePage.js [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+            wins: el.numberOfTimesWhenArchetypeWon
+          }
+        }),
+
+        gamesTotal: gamesTotal,
+        gamesWon: gamesWon,
+        firstPlayed: this.prettifyDate(res.data.whenDeckWasFirstPlayed)
+      }
+      this.setState({
+        deck: newDeckState,
```
Comment (NightAngell) [inline src/Pages/ArchetypePage/ArchetypePage.js]: deck or archetype here?

File: src/Pages/ArchetypePage/ArchetypePage.js [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+  goToPreviousPage() {
+    this.props.history.goBack();
+  }
+
+  render() {
+    this.checkIfProperDeckId();
+
+    const cardsListContent = this.createCardsTable(this.state.deck.cardsList);
+
+
```
Comment (NightAngell) [inline src/Pages/ArchetypePage/ArchetypePage.js]: Too many empty space here.

File: src/Pages/ArchetypePage/ArchetypePage.js [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+        }
+      });
+      if(cards.length > 0) {
+        let row = [];
+        row.push(<th key={type + "name"}>{this.prietifyCardTypeString(type)}</th>);
+        row.push(<th key={type + "content"}>{cards}</th>);
+        cardTable.push(<tr key={type}>{row}</tr>);
+      }
+    });
+    return cardTable;
```
Comment (NightAngell) [inline src/Pages/ArchetypePage/ArchetypePage.js:110]: Empty space before return.

File: src/Pages/DeckList/DeckList.js [hunk 14]
```diff
@@ -39,11 +88,38 @@ class DeckList extends Component {
...
+
+    getLoactionParameters() {
+        const locationParameters = this.props.location.search.split('?');
+        let locationConcreteValues = {};
+        locationParameters.forEach(param => {
+            let nameAndValue = param.split("=");
+            if(nameAndValue[0])
+                locationConcreteValues[nameAndValue[0]] = nameAndValue[1];
+        });
+        return locationConcreteValues;
```
Comment (NightAngell) [inline src/Pages/DeckList/DeckList.js:110]: Empty space before return.

File: src/Pages/DeckList/DeckList.js [hunk 15]
```diff
@@ -39,11 +88,38 @@ class DeckList extends Component {
...
     getPageNumber() {
         if(!this.props.location.search) return 1;
-        else return Number(this.props.location.search.split('=')[1]);
+
+        const locationParameters = this.getLoactionParameters();
+        if(locationParameters.pageNumber)
+        {
+            return Number(locationParameters.pageNumber);
+        }
+        return 1
```
Comment (NightAngell) [inline src/Pages/DeckList/DeckList.js:98]: Empty space before return.

File: src/Pages/DeckPage/DeckPage.js [hunk 16]
```diff
@@ -1,147 +1,205 @@
...
-      array.forEach(el => {
-        jsxArray.push((<div className="card" key={el}>{el}</div>));
+  //changes string like "loremIpsumDolor" into "Lorem Ipsum Dolor" (adds spaces and makes first letter uppercased)
+  prietifyCardTypeString(inputString) {
+    let newString = inputString[0].toUpperCase();
+    for (let i = 1; i < inputString.length; i++) {
+      if(inputString[i] === inputString[i].toUpperCase()) newString += " ";
+      newString += inputString[i];
+    }
+    return newString;
```
Comment (NightAngell) [inline src/Pages/DeckPage/DeckPage.js:86]: Empty space before return.

File: src/Pages/DeckPage/DeckPage.js [hunk 17]
```diff
@@ -1,147 +1,205 @@
...
       });
-      return(jsxArray);
+      if(cards.length > 0) {
+        let row = [];
+        row.push(<th key={type + "name"}>{this.prietifyCardTypeString(type)}</th>);
+        row.push(<th key={type + "content"}>{cards}</th>);
+        cardTable.push(<tr key={type}>{row}</tr>);
+      }
+    });
+    return cardTable;
```
Comment (NightAngell) [inline src/Pages/DeckPage/DeckPage.js:117]: Empty space before return.

**Review by NightAngell — CHANGES_REQUESTED** [review]
In many places code is just copy and pasted (not reusable), but due to deadline we have no time to change this. At least remove (or add) empty space in requested places.

File: src/Pages/ArchetypeList/ArchetypeList.js (hunk 8 above)
Comment (Erakedon) [reply src/Pages/ArchetypeList/ArchetypeList.js]: This part shouldn't be commented. I'm uncommenting it.

**Review by NightAngell — APPROVED** [review]

**Pull request merged**